### PR TITLE
upgraded migration script

### PR DESCRIPTION
**Pull request by TVTaster**
Improved with the follows:
-  `poetry` for package management, versioning, deployment
-  `loguru` for more robust logging
-  `pydantic` for strict modeling
-   multiple domains support
-   type hints bulletproofish ;)

### CODE REVIEW

File: mailgun-template-migration.py [hunk 1]
```diff
@@ -1,75 +1,95 @@
...
+        logger.error(f" template {template.name} cannot be stored in {env_to_copy.name} environment")
     else:
-        migrate_template(sys.argv[1])
+        logger.info(f" template {template.name} stored successfully in {env_to_copy.name} environment")
+
+
+def get_all_templates() -> List[BaseTemplate]:
+    templates: List[BaseTemplate] = []
+    logger.info(f"Fetching all templates under domain {MG_TEMPLATES_DOMAIN}")
+    templates_url = f"{MG_BASE_URL}/{MG_TEMPLATES_DOMAIN}/templates"
```
Comment (yaarams) [inline mailgun-template-migration.py]: extract to global variable

File: mailgun-template-migration.py [hunk 2]
```diff
@@ -1,75 +1,95 @@
...
+def read_arguments() -> Namespace:
+    parser = ArgumentParser(description='MailGun template migrator')
+    parser.add_argument('action', nargs='?', type=Action, default=Action.COPY,
+                        help='choose which action would you like to do. only *copy* supported')
+    parser.add_argument('env', type=Env, default=Env.DEV,
+                        help='choose which environment to copy templates from. local, dev, prod or ALL')
+    return parser.parse_args()
+
+
+def get_template(templates: List[BaseTemplate]) -> List[FullTemplate]:
```
Comment (yaarams) [inline mailgun-template-migration.py]: get_template_content (why not place the function in class?)
why not do this fetch as part of get_templates?

File: mailgun-template-migration.py [hunk 3]
```diff
@@ -1,75 +1,95 @@
...
+    return full_templates
+
+
+def mail_domain_resolver(env: Env) -> str:
+    return {Env.LOCAL: MG_LOCAL_MAIL_DOMAIN,
+            Env.DEV: MG_DEV_MAIL_DOMAIN,
+            Env.PROD: MG_PROD_MAIL_DOMAIN}.get(env)
+
+
+def create_template(template: FullTemplate, env_to_copy: Env):
```
Comment (yaarams) [inline mailgun-template-migration.py]: create_template_in_env

File: mailgun-template-migration.py (hunk 1 above)
Comment (TVTaster) [reply mailgun-template-migration.py]: done :)

File: mailgun-template-migration.py (hunk 2 above)
Comment (TVTaster) [reply mailgun-template-migration.py]: done :)

File: mailgun-template-migration.py (hunk 3 above)
Comment (TVTaster) [reply mailgun-template-migration.py]: done :)

**Pull request merged**